services/librariesio/librariesio-base.js

```diff
@@ -0,0 +1,39 @@
+import Joi from 'joi'
+import { anyInteger, nonNegativeInteger } from '../validators.js'
+import { BaseJsonService } from '../index.js'
+
+// API doc: https://libraries.io/api#project
+const projectSchema = Joi.object({
+  platform: Joi.string().required(),
+  dependents_count: nonNegativeInteger,
+  dependent_repos_count: nonNegativeInteger,
+  rank: anyInteger,
+}).required()
+
+function createRequestFetcher(context, config) {
+  const { sendAndCacheRequest, librariesIoApiProvider } = context
+
+  return async (url, options) =>
+    await librariesIoApiProvider.requestAsPromise(
+      sendAndCacheRequest,
+      url,
+      options
+    )
+}
+
+export default class LibrariesIoBase extends BaseJsonService {
+  constructor(context, config) {
+    super(context, config)
+    this._requestFetcher = createRequestFetcher(context, config)
+  }
+
+  async fetchProject({ platform, scope, packageName }) {
+    return this._requestJson({
+      schema: projectSchema,
+      url: `/${encodeURIComponent(platform)}/${
+        scope ? encodeURIComponent(`${scope}/`) : ''
+      }${encodeURIComponent(packageName)}`,
+      errorMessages: { 404: 'package not found' },
+    })
+  }
+}
```

Comment:
I personally tend to prefer the reusable helper/fetcher type functions over extending the inheritance hierarchy, but it felt cleaner/more straightforward to me to use a base class for setting up the request fetcher on construction in a base class.